.pipelines/build-linux-valgrind.yml

```diff
@@ -17,3 +17,4 @@ steps:
   displayName: Build C++ VW binary
 - bash: ./.scripts/linux/test-valgrind.sh
   displayName: Run test suite with Valgrind
+  failOnStderr: true
```

Comment:
why don't we need this on the other files, like the java one? is it because this one returns a 0 error code even if it failed